onadata/apps/api/models/odk_token.py

```diff
@@ -99,5 +94,16 @@ def _post_save_persist_partial_digests(sender, instance=None, **kwargs):
         _persist_partial_digests(instance.user)
 
 
+def _post_save_set_expiry_date(sender, instance=None, **kwargs):
+    if instance and not instance.expires:
+        instance.expires = instance.created + timedelta(
+            days=ODK_TOKEN_LIFETIME)
+        instance.expires = instance.expires.astimezone(
+            instance.created.tzinfo)
```

Comment:
Why are we overriding `instance.expires` in line 101? Also, will `instance.expires` in line 99 still have the value assigned to it even though it hasn't been saved and it's being used in line 101? Let me know if my questions make sense

Reply:
Overriding the value to make sure that the datetime object saved in the field is timezone aware(_Usually Django saves it's `datetime` objects with the UTC timezone when Timezones are enabled. Didn't want to deviate from that_)

The instance will still have the `expires` value assigned to the object in line 99 and the value will be accessible in line 101, I believe. I'll update this though and set the `expires` field in one line.